kso_utils/general.py

```diff
@@ -21,3 +21,18 @@ def parallel_map(func, iterable, args=()):
     with multiprocessing.Pool() as pool:
         results = pool.starmap(func, zip(iterable, *args))
     return results
+
+
+def validate_utils(utils, required_functions):
```

Comment:
This is quite a good idea, but this shouldn't be done at runtime, this should be a test that iterates over the modules in `kso_utils.registries`  and checks that they all implement the required interfaces